### ISSUE TITLE
[Bug]: Unhandled rejection:  Promise {icrosoft/1ds-post-js: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms   <rejected> Error: Unknown platform: android

### ISSUE THREAD

**Issue by Mahoo12138** [coder/code-server#5593]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS/Web Information

- Web Browser: Edge 105.0.1343.50 (Official build) (64-bit)
- Local OS: Windows 10 Pro 21H2 19044.2006
- Remote OS: Android 12 4.19.255-pe-g631678dbd52f
- Remote Architecture: aarch64
- `code-server --version`: 4.7.0

### Steps to Reproduce

1.  Follow the [NPM Installation Docs](https://coder.com/docs/code-server/latest/termux#npm-installation)
2. Launch the `code-server` in terminal
3. Then,  stop it and exec the upgrade command: `npm update --global code-server --unsafe-perm`

### Expected

No errors and exceptions

### Actual

```
...
> @vscode/ripgrep@1.14.2 postinstall
> node ./lib/postinstall.js

Unhandled rejection:  Promise {icrosoft/1ds-post-js: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
  <rejected> Error: Unknown platform: android
      at getTarget (/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/@vscode/ripgrep/lib/postinstall.js:60:24)
      at main (/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/@vscode/ripgrep/lib/postinstall.js:78:23)
} reason: Error: Unknown platform: android
    at getTarget (/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/@vscode/ripgrep/lib/postinstall.js:60:24)
    at main (/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/@vscode/ripgrep/lib/postinstall.js:78:23)
[##################] \ reify:@microsoft/1ds-post-js: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
> node-pty@0.11.0-beta11 postinstall
> node scripts/post-install.js

[##################] | reify:array-flatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
added 131 packages in 1m

10 packages are looking for funding
  run `npm fund` for details
...
```

### Logs

```
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.
[##################] - reify:array-flatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
> argon2@0.29.1 install
> node-pre-gyp install --fallback-to-build

node-pre-gyp info it worked if it ends with oktp fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
node-pre-gyp info using node-pre-gyp@1.0.9
node-pre-gyp info using node@16.16.0 | android | arm64
node-pre-gyp info check checked for "/data/data/com.termux/files/usr/lib/node_modules/code-server/node_modules/argon2/lib/binding/napi-v3/argon2.node" (not found)
node-pre-gyp http GET https://github.com/ranisalt/node-argon2/releases/download/v0.29.1/argon2-v0.29.1-napi-v3-android-arm64-unknown.tar.gz
node-pre-gyp ERR! install request to https://github.com/ranisalt/node-argon2/releases/download/v0.29.1/argon2-v0.29.1-napi-v3-android-arm64-unknown.tar.gz failed, reason: connect ETIMEDOUT 20.205.243.166:443 
node-pre-gyp WARN Pre-built binaries not installable for argon2@0.29.1 and node@16.16.0 (node-v93 ABI, unknown) (falling back to source compile with node-gyp) 
node-pre-gyp WARN Hit error request to https://github.com/ranisalt/node-argon2/releases/download/v0.29.1/argon2-v0.29.1-napi-v3-android-arm64-unknown.tar.gz failed, reason: connect ETIMEDOUT 20.205.243.166:443 
gyp info it worked if it ends with okatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
gyp info using node-gyp@9.0.0
gyp info using node@16.16.0 | android | arm64
gyp info ok #######] | reify:array-flatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
gyp info it worked if it ends with okatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
gyp info using node-gyp@9.0.0array-flatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
gyp info using node@16.16.0 | android | arm64
gyp info find Python using Python version 3.10.7 found at "/data/data/com.termux/files/usr/bin/python3"-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
gyp info spawn /data/data/com.termux/files/usr/bin/python3 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
gyp info spawn args [
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/code-server/node_modules/argon2/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/home/.cache/node-gyp/16.16.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/data/data/com.termux/files/home/.cache/node-gyp/16.16.0',
gyp info spawn args   '-Dnode_gyp_dir=/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/data/data/com.termux/files/home/.cache/node-gyp/16.16.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/data/data/com.termux/files/usr/lib/node_modules/code-server/node_modules/argon2',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info ok #######] | reify:array-flatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
gyp info it worked if it ends with okatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
gyp info using node-gyp@9.0.0
gyp info using node@16.16.0 | android | arm64
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/data/data/com.termux/files/usr/lib/node_modules/code-server/node_modules/argon2/build'.1.1/array-flatten-2.1.1.tgz 4267
  CC(target) Release/obj.target/libargon2/argon2/src/ref.o
  CC(target) Release/obj.target/libargon2/argon2/src/argon2.o0 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  CC(target) Release/obj.target/libargon2/argon2/src/core.o200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  CC(target) Release/obj.target/libargon2/argon2/src/blake2/blake2b.o//cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  CC(target) Release/obj.target/libargon2/argon2/src/thread.o0 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  CC(target) Release/obj.target/libargon2/argon2/src/encoding.ohttps://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  AR(target) Release/obj.target/argon2.aen: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  COPY Release/argon2.areify:array-flatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  CXX(target) Release/obj.target/argon2/src/argon2_node.oT 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  SOLINK_MODULE(target) Release/obj.target/argon2.node GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  COPY Release/argon2.nodefy:array-flatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  COPY /data/data/com.termux/files/usr/lib/node_modules/code-server/node_modules/argon2/lib/binding/napi-v3/argon2.node
  TOUCH Release/obj.target/action_after_build.stamptch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
make: Leaving directory '/data/data/com.termux/files/usr/lib/node_modules/code-server/node_modules/argon2/build'
gyp info ok 
node-pre-gyp info ok 
[##################] | reify:array-flatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
> code-server@4.7.0 postinstall
> sh ./postinstall.sh

curl: (28) Failed to connect to github.com port 443 after 31241 ms: Connection timed out/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
Failed to download cloud agent; --link will not work
Installing Code dependencies...
[##################] \ reify:@microsoft/applicationinsights-core-js: timing reifyNode:node_modules/@microsoft/1ds-core-js Completed in 3850ms
> @parcel/watcher@2.0.5 install
> node-gyp-build

gyp info it worked if it ends with okft/1ds-post-js: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
gyp info using node-gyp@9.0.0
gyp info using node@16.16.0 | android | arm64
gyp info find Python using Python version 3.10.7 found at "/data/data/com.termux/files/usr/bin/python3"oninsights-core-js Completed in 3857ms
gyp info spawn /data/data/com.termux/files/usr/bin/python3 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
gyp info spawn args [\ reify:@microsoft/1ds-post-js: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/@parcel/watcher/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/home/.cache/node-gyp/16.16.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/data/data/com.termux/files/home/.cache/node-gyp/16.16.0',
gyp info spawn args   '-Dnode_gyp_dir=/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/data/data/com.termux/files/home/.cache/node-gyp/16.16.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/@parcel/watcher',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make] | reify:array-flatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/@parcel/watcher/build'
  CC(target) Release/obj.target/nothing/../../node-addon-api/nothing.o/cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  AR(target) Release/obj.target/../../node-addon-api/nothing.a https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  COPY Release/nothing.aeify:array-flatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  CXX(target) Release/obj.target/watcher/src/binding.oiming reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
  CXX(target) Release/obj.target/watcher/src/Watcher.oiming reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
  CXX(target) Release/obj.target/watcher/src/Backend.o GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  CXX(target) Release/obj.target/watcher/src/DirTree.o GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  SOLINK_MODULE(target) Release/obj.target/watcher.nodeGET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  COPY Release/watcher.nodey:array-flatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
make: Leaving directory '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/@parcel/watcher/build'2.1.1.tgz 4267
gyp info ok 

> keytar@7.9.0 install
> prebuild-install || npm run build

[##################] | reify:array-flatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
> keytar@7.9.0 build
> node-gyp rebuild

gyp info it worked if it ends with okatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
gyp info using node-gyp@9.0.0
gyp info using node@16.16.0 | android | arm64
gyp info find Python using Python version 3.10.7 found at "/data/data/com.termux/files/usr/bin/python3"oninsights-core-js Completed in 3857ms
gyp info spawn /data/data/com.termux/files/usr/bin/python3 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
gyp info spawn args [
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/keytar/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/home/.cache/node-gyp/16.16.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/data/data/com.termux/files/home/.cache/node-gyp/16.16.0',
gyp info spawn args   '-Dnode_gyp_dir=/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/data/data/com.termux/files/home/.cache/node-gyp/16.16.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/keytar',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make] | reify:array-flatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/keytar/build'flatten-2.1.1.tgz 4267
  CXX(target) Release/obj.target/keytar/src/async.o
  CXX(target) Release/obj.target/keytar/src/main.os: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
  CXX(target) Release/obj.target/keytar/src/keytar_posix.o 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  SOLINK_MODULE(target) Release/obj.target/keytar.node GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
  COPY Release/keytar.nodefy:array-flatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
make: Leaving directory '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/keytar/build'-flatten-2.1.1.tgz 4267
gyp info ok 

> native-watchdog@1.4.0 install
> node-gyp rebuild

gyp info it worked if it ends with okft/1ds-post-js: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
gyp info using node-gyp@9.0.0
gyp info using node@16.16.0 | android | arm64
gyp info find Python using Python version 3.10.7 found at "/data/data/com.termux/files/usr/bin/python3"-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
gyp info spawn /data/data/com.termux/files/usr/bin/python3g reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
gyp info spawn args [
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/native-watchdog/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/home/.cache/node-gyp/16.16.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/data/data/com.termux/files/home/.cache/node-gyp/16.16.0',
gyp info spawn args   '-Dnode_gyp_dir=/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/data/data/com.termux/files/home/.cache/node-gyp/16.16.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/native-watchdog',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make] / reify:@microsoft/1ds-post-js: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/native-watchdog/build'
  CXX(target) Release/obj.target/watchdog/src/watchdog.oing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
  SOLINK_MODULE(target) Release/obj.target/watchdog.nodeing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
  COPY Release/watchdog.node:@microsoft/1ds-post-js: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
make: Leaving directory '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/native-watchdog/build'in 3857ms
gyp info ok 

> node-pty@0.11.0-beta11 install
> node scripts/install.js

gyp info it worked if it ends with okft/1ds-post-js: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
gyp info using node-gyp@9.0.0
gyp info using node@16.16.0 | android | arm64
gyp info find Python using Python version 3.10.7 found at "/data/data/com.termux/files/usr/bin/python3"oninsights-core-js Completed in 3857ms
gyp info spawn /data/data/com.termux/files/usr/bin/python3g reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
gyp info spawn args [
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/node-pty/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/home/.cache/node-gyp/16.16.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/data/data/com.termux/files/home/.cache/node-gyp/16.16.0',
gyp info spawn args   '-Dnode_gyp_dir=/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/data/data/com.termux/files/home/.cache/node-gyp/16.16.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/node-pty',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make] | reify:@microsoft/1ds-post-js: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/node-pty/build'
  CXX(target) Release/obj.target/pty/src/unix/pty.o
  SOLINK_MODULE(target) Release/obj.target/pty.node: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
  COPY Release/pty.nodereify:@microsoft/1ds-post-js: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
make: Leaving directory '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/node-pty/build'
gyp info ok 
[##################] | reify:@microsoft/1ds-post-js: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
> spdlog@0.13.6 install
> node-gyp rebuild

gyp info it worked if it ends with okatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
gyp info using node-gyp@9.0.0@microsoft/1ds-post-js: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
gyp info using node@16.16.0 | android | arm64
gyp info find Python using Python version 3.10.7 found at "/data/data/com.termux/files/usr/bin/python3"oninsights-core-js Completed in 3857ms
gyp info spawn /data/data/com.termux/files/usr/bin/python3g reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
gyp info spawn args [
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/spdlog/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/data/data/com.termux/files/home/.cache/node-gyp/16.16.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/data/data/com.termux/files/home/.cache/node-gyp/16.16.0',
gyp info spawn args   '-Dnode_gyp_dir=/data/data/com.termux/files/usr/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/data/data/com.termux/files/home/.cache/node-gyp/16.16.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/spdlog',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make] \ reify:@microsoft/1ds-post-js: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
make: Entering directory '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/spdlog/build'
  CXX(target) Release/obj.target/spdlog/src/main.o
  CXX(target) Release/obj.target/spdlog/src/logger.o timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
  SOLINK_MODULE(target) Release/obj.target/spdlog.nodeiming reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
  COPY Release/spdlog.nodefy:@microsoft/1ds-post-js: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
make: Leaving directory '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/spdlog/build'ompleted in 3857ms
gyp info ok 

> @vscode/ripgrep@1.14.2 postinstall
> node ./lib/postinstall.js

Unhandled rejection:  Promise {icrosoft/1ds-post-js: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
  <rejected> Error: Unknown platform: android
      at getTarget (/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/@vscode/ripgrep/lib/postinstall.js:60:24)
      at main (/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/@vscode/ripgrep/lib/postinstall.js:78:23)
} reason: Error: Unknown platform: android
    at getTarget (/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/@vscode/ripgrep/lib/postinstall.js:60:24)
    at main (/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/node_modules/@vscode/ripgrep/lib/postinstall.js:78:23)
[##################] \ reify:@microsoft/1ds-post-js: timing reifyNode:node_modules/@microsoft/applicationinsights-core-js Completed in 3857ms
> node-pty@0.11.0-beta11 postinstall
> node scripts/post-install.js

[##################] | reify:array-flatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
added 131 packages in 1m

10 packages are looking for funding
  run `npm fund` for details

> vscode-extensions@0.0.1 postinstall
> node ./postinstall.mjs

Removed /data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/extensions/node_modules/typescript/AUTHORS.mdlatten-2.1.1.tgz 4267
Removed /data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/extensions/node_modules/typescript/CODE_OF_CONDUCT.md
Removed /data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/extensions/node_modules/typescript/CopyrightNotice.txt
Removed /data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/extensions/node_modules/typescript/LICENSE.txt
Removed /data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/extensions/node_modules/typescript/README.md
Removed /data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/extensions/node_modules/typescript/SECURITY.md
Removed /data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/extensions/node_modules/typescript/ThirdPartyNoticeText.txt
Removed /data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/extensions/node_modules/typescript/bin
Removed /data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/extensions/node_modules/typescript/loc
removed '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/extensions/node_modules/typescript/lib/tsc.js'tten-2.1.1.tgz 4267
removed '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/extensions/node_modules/typescript/lib/tsserverlibrary.d.ts'
removed '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/extensions/node_modules/typescript/lib/tsserverlibrary.js'
removed '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/extensions/node_modules/typescript/lib/typescriptServices.d.ts'
removed '/data/data/com.termux/files/usr/lib/node_modules/code-server/lib/vscode/extensions/node_modules/typescript/lib/typescriptServices.js'

added 1 package in 2s
[##################] | reify:array-flatten: http fetch GET 200 https://cdn.npmmirror.com/packages/array-flatten/2.1.1/array-flatten-2.1.1.tgz 4267
> protobufjs@6.11.3 postinstall
> node scripts/postinstall


added 74 packages, and changed 210 packages in 3m

22 packages are looking for funding
  run `npm fund` for details
```

### Does this issue happen in VS Code or GitHub Codespaces?

- [X] I cannot reproduce this in VS Code.
- [X] I cannot reproduce this in GitHub Codespaces.

### Are you accessing code-server over HTTPS?

- [X] I am using HTTPS.

**Comment by code-asher** [comment]
Is `aarch64` the same as `arm64`?  Maybe setting `npm_config_arch` to `arch64` could do it since that seems to be in their list of supported architectures.  If they are the same thing then this could be considered an upstream bug with `@vscode/ripgrep`.

**Comment by T-Dynamos** [comment]
This should be closed now as #5745

**Comment by Mahoo12138** [comment]
> This should be closed now as #5745

Amazing!
